### PR TITLE
Update 0009-metric-handles to match current spec

### PR DESCRIPTION
**Pull request by jmacd**
This OTEP reflected a relatively-complicated association between instruments, their recommended keys, and handles. Much of this text was no longer needed. In particular, a large "Trade-offs and mitigations" block of text was completely rewritten as 0049-metric-label-set.md.

This brings this OTEP into line with the current spec. I believe there are no significant discrepancies between this and the current spec. In a future OTEP I will propose the renaming of "Handle" to "Bound instrument". Note, however, that "Handle" is not a recommended type name, it is only a concept. Each kind of instrument has a distinct type of handle, but in discussing this feature, we will use "Bound instrument" in the future.

### CODE REVIEW

File: text/0009-metric-handles.md [hunk 1]
```diff
@@ -1,74 +1,36 @@
...
+Handles are useful to reduce the cost of repeatedly recording a metric instrument (cumulative, gauge, or measure) with a pre-defined set of label values.
 
-`GetHandle` specifies two calling patterns that may be supported: one with ordered label values and one without.  The facility for ordered label values is provided as a potential optimization that facilitates a simple lookup for the SDK; in this form, the API is permitted to thrown an exception or return an error when there is a mismatch in the arguments to `GetHandle`.  When label values are accepted in any order, some SDKs may perform an expensive lookup to find an existing metrics handle, but they must not throw exceptions.
+`GetHandle` gets a new handle given a [`LabelSet`](./0049-metric-label-set.md).
```
Comment (lzchen) [inline text/0009-metric-handles.md]: This link doesn't seem to work.

File: text/0009-metric-handles.md [hunk 2]
```diff
@@ -1,74 +1,36 @@
...
 
 Applications are encouraged to re-use metric handles for efficiency.
 
-Handles are useful to reduce the cost of repeatedly recording a metric instrument (cumulative, gauge, or measure) with a pre-defined set of label values.  All metric kinds support declaring a set of required label keys.  These label keys, by definition, must be specified in every metric `Handle`.  We permit "unspecified" label values in cases where a handle is requested but a value was not provided.  The default metric handle has all its required keys unspecified.  We presume that fast pre-aggregation of metrics data is only possible, in general, when the pre-aggregation keys are a subset of the required keys on the metric.
+Handles are useful to reduce the cost of repeatedly recording a metric instrument (cumulative, gauge, or measure) with a pre-defined set of label values.
```
Comment (lzchen) [inline text/0009-metric-handles.md]: Should we stick with `LabelSet` everywhere that mentions the concept of labels? There seems to be a lot of confusion from the difference between that and Label Keys/Values.

File: text/0009-metric-handles.md (hunk 1 above)
Comment (jmacd) [reply text/0009-metric-handles.md]: This is in PR #49.

File: text/0009-metric-handles.md (hunk 2 above)
Comment (jmacd) [reply text/0009-metric-handles.md]: LabelSet wasn't introduced at the time of this OTEP. This is why revising OTEPs is difficult. LabelSet is not introduced yet, although I put in the link to 0049.

File: text/0009-metric-handles.md [hunk 3]
```diff
@@ -1,74 +1,36 @@
...
+`GetHandle` gets a new handle given a [`LabelSet`](./0049-metric-label-set.md).
 
-`GetHandle` and `GetDefaultHandle` support additional label values not required in the definition of the metric instrument.  These optional labels act the same as pre-defined labels in the low-level metrics data representation, only that they are not required.  Some SDKs may elect to use additional label values as the "attachment" data on metrics.
+As a language-optional feature, the API may provide an _ordered_ form of the API for supplying labels in known order.  The ordered label-value API is provided as a (language-optional) potential optimization that facilitates a simple lookup for the SDK.  In this ordered-value form, the API is permitted to throw an exception or return an error when there is a mismatch in the arguments to `GetHandle`, although languages without strong type-checking may wish to omit this feature.  When label values are accepted in any order, SDKs may be forced to canonicalize the labels in order to find an existing metrics handle, but they must not throw exceptions.
```
Comment (lzchen) [inline text/0009-metric-handles.md]: Same here. Are these still relevant if we the only way to provide labels is through `LabelSet`?

Comment (jmacd) [reply text/0009-metric-handles.md]: It's still an optional feature, also discussed int the spec. I don't care for this feature, but it is a potential optimization.

File: text/0009-metric-handles.md [hunk 4]
```diff
@@ -1,74 +1,36 @@
...
-**Status:** `proposed`
+**Status:** `accepted`
 
-Specify the behavior of the Metrics API `Handle` type, for efficient repeated-use of metric instruments.
+Specify the behavior of the Metrics API "Handle" type, for efficient repeated-use of metric instruments.
 
 ## Motivation
 
-The specification currently names this concept `TimeSeries`, the object returned by `GetOrCreateTimeseries`, which supports binding a metric to a pre-defined set of labels for repeated use.  This proposal renames these `Handle` and `GetHandle`, respectively, and adds further detail to the API specification for handles.  
+The specification currently names this concept "TimeSeries", the type returned by `GetOrCreateTimeseries`, which supports binding a metric to a pre-defined set of labels for repeated use.  This proposal renames these "Handle" and `GetHandle`, respectively, and adds further detail to the API specification for handles.
```
Comment (bogdandrutu) [inline text/0009-metric-handles.md]: I think you are using `acquireHandle` in Go. Is that a better name?

Comment (jmacd) [reply text/0009-metric-handles.md]: True. If you'd like me to continue updating this OTEP inside of this PR, I will. The specification says "GetHandle" currently.

During review of the current specification, Yuri requested that we renamed handles to "Bound instruments" and "GetHandle" to "Bind". I was hoping to put this proposal in a future OTEP, combined with a specification for a "Release" method to free a binding.

Comment (bogdandrutu) [reply text/0009-metric-handles.md]: ACK.

**Comment by c24t** [comment]
For anyone else struggling to keep up with metrics work: https://github.com/open-telemetry/opentelemetry-specification/pull/299 made significant changes to the metrics spec that weren't yet described by OTEP 0009. The specs PR was informed by go prototypes at https://github.com/open-telemetry/opentelemetry-go/pull/100 and https://github.com/open-telemetry/opentelemetry-go/pull/172.

The specs changes were merged, but the OTEP was still "proposed" (i.e. not yet "accepted"). In the meantime https://github.com/open-telemetry/oteps/pull/49 was merged, and the process was changed (in https://github.com/open-telemetry/oteps/pull/55) so that OTEPs wouldn't be merged until they were "accepted". This PR revises the OTEP and moves it to "accepted". Usually the next step would be to edit the spec to reflect the OTEP, but in this case that's already done.

**Comment by jmacd** [comment]
Thank you. I will merge this tomorrow.

**Comment by jmacd** [comment]
This has the required number of approvals.

**Pull request merged**